test/MatrixFields/rmul_with_projection.jl

```diff
@@ -18,13 +23,13 @@ function test_rmul_with_projection(x::X, y::Y, lg, expected_result) where {X, Y}
         zip(object2tuple(result), object2tuple(expected_result)),
     )
 
-    @test max_error <= 1                                   # correctness
-    @test (@allocated rmul_with_projection(x, y, lg)) == 0 # allocations
-    @test_opt rmul_with_projection(x, y, lg)               # type instabilities
+    @test max_error <= 1                                                     # correctness
+    @test (@allocated rmul_with_projection(x, y, lg)) == 0                   # allocations
+    @test_opt rmul_with_projection(x, y, lg)                                 # type instabilities
 
-    @test result_type == typeof(result)                    # correctness
-    @test (@allocated rmul_return_type(X, Y)) == 0         # allocations
-    @test_opt rmul_return_type(X, Y)                       # type instabilities
+    @test result_type == typeof(result)                                      # correctness
+    @test (@allocated Base.promote_op(rmul_with_projection, X, Y, LG)) == 0  # allocations
+    @test_opt Base.promote_op(rmul_with_projection, X, Y, LG)                # type instabilities
```

Comment:
I think we can drop these tests if `@test_opt rmul_with_projection(x, y, lg)` passes.